docs/server/source/production-deployment-template-tendermint/workflow.rst

```diff
@@ -0,0 +1,188 @@
+
+☐ Name of the NGINX instance (``ngx-http-instance-*`` or ``ngx-https-instance-*``)
+
+☐ Name of the OpenResty instance (``openresty-instance-*``)
+
+☐ Name of the MongoDB monitoring agent instance (``mdb-mon-instance-*``)
+
+☐ Name of the Tendermint instance (``tendermint-instance-*``)
+
+Example
+^^^^^^^
+
+.. code:: text
+
+  {
+    "MongoDB": [
+      "mdb-instance-1",
+      "mdb-instance-2",
+      "mdb-instance-3",
+      "mdb-instance-4"
+    ],
+    "BigchainDB": [
+      "bdb-instance-1",
+      "bdb-instance-2",
+      "bdb-instance-3",
+      "bdb-instance-4"
+    ],
+    "NGINX": [
+      "ngx-instance-1",
+      "ngx-instance-2",
+      "ngx-instance-3",
+      "ngx-instance-4"
+    ],
+    "OpenResty": [
+      "openresty-instance-1",
+      "openresty-instance-2",
+      "openresty-instance-3",
+      "openresty-instance-4"
+    ],
+    "MongoDB_Monitoring_Agent": [
+      "mdb-mon-instance-1",
+      "mdb-mon-instance-2",
+      "mdb-mon-instance-3",
+      "mdb-mon-instance-4"
+    ],
+    "Tendermint": [
+      "tendermint-instance-1",
+      "tendermint-instance-2",
+      "tendermint-instance-3",
+      "tendermint-instance-4"
+    ]
+  }
+
+
+☐ Generate three keys and corresponding certificate signing requests (CSRs):
+
+#. Server Certificate for the MongoDB instance
+#. Client Certificate for BigchainDB Server to identify itself to MongoDB
+#. Client Certificate for MongoDB Monitoring Agent to identify itself to MongoDB
+
+Use the self-signed CA to sign those three CSRs:
+
+* Three certificates (one for each CSR).
+
+For help, see the pages:
+
+* :doc:`How to Generate a Server Certificate for MongoDB <../production-deployment-template/server-tls-certificate>`
+* :doc:`How to Generate a Client Certificate for MongoDB <../production-deployment-template/client-tls-certificate>`
+
+☐ Make up an FQDN for your BigchainDB node (e.g. ``mynode.mycorp.com``).
+Make sure you've registered the associated domain name (e.g. ``mycorp.com``),
+and have an SSL certificate for the FQDN.
+(You can get an SSL certificate from any SSL certificate provider.)
+
+☐ Ask the managing organization for the user name to use for authenticating to
+MongoDB.
+
+☐ If the cluster uses 3scale for API authentication, monitoring and billing,
+you must ask the managing organization for all relevant 3scale credentials -
+secret token, service ID, version header and API service token.
+
+☐ If the cluster uses MongoDB Cloud Manager for monitoring,
+you must ask the managing organization for the ``Project ID`` and the
+``Agent API Key``.
+(Each Cloud Manager "Project" has its own ``Project ID``. A ``Project ID`` can
+contain a number of ``Agent API Key`` s. It can be found under
+**Settings**. It was recently added to the Cloud Manager to
+allow easier periodic rotation of the ``Agent API Key`` with a constant
+``Project ID``)
+
+
+.. _generate-the-blockchain-id-and-genesis-time:
+
+3. Generate the Blockchain ID and Genesis Time
```

Comment:
This is something that the cluster/blockchain "managing organization" must do first, so it should go right after item 2. above, i.e. if h2 means a level-two heading, then the sections and headings should be:

h2 Things the Managing Organization Must Do First

h3 1. Set Up a Self-Signed Certificate Authority

h3 2. Register a Domain and Get an SSL Certificate for It

h3 3. Generate the Blockchain ID and Genesis Time

h2 Things Each Node Operator Must Do

☐ Checkbox thing 1

☐ Checkbox thing 2

☐ etc.

Reply:
Actually, now that I think about it more, step 1 (Set Up a Self-Signed Certificate Authority) is no longer something that needs to be done at the cluster/blockchain level (i.e. by the "managing organization"). It's now something that each node operator can do independently, so it should be moved under "h2 Things Each Node Operator Must Do" (i.e. with a checkbox ☐ in front).

Reply:
Done.

Reply:
:+1:  + I did a small commit just now (on this PR, below) to add a heading before 1. and 2. It says **Things to Do Before Deploying Any Nodes**.